### DISCUSSION TITLE
What are the advantages of RavenDB ?

### DISCUSSION THREAD

**Discussion by sgf** [ravendb/ravendb#19488] [Q&A] [answered]
What is the query performance of RavenDB for the same data? 
What is the indexing speed? 
What about CPU usage? 
What is the memory use? 
What is the size of the index file?

Can RavenDB be used as an alternative to distributed search such as elasticsearch?

The .NET ecosystem is still a blank slate. There isn't any solution available. lucenet.net development is very slow and backward.

**Comment by ayende** [comment, marked as answer]
Hi,
We publish some benchmarks here that can give you some idea about the performance on various types of machines:
https://ravendb.net/performance

Yes, RavenDB can be used as a distributed & sharded search engine. 

Note that RavenDB comes with both Lucene.NET indexing engine and the Corax indexing engine (built specifically for RavenDB).
Corax is about 10x faster than the Lucene version, and the RavenDB's version of Lucene was heavily optimized (faster than the normal one)

**Comment by DejanMilicic** [comment]
Yes, you can use RavenDB as an alternative to ElasticSearch.
The same feature set is available (standard and custom analyzers, wildcards, highlights, suggestions, spatial searching, etc.)
